tools/lambda-promtail/lambda-promtail/cw_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/pkg/logproto"
+)
+
+func Test_parseCWEvent(t *testing.T) {
+	tests := []struct {
+		name           string
+		b              *batch
+		expectedStream string
+		keepStream     bool
+	}{
+		{
+			name: "cloudwatch",
+			b: &batch{
+				streams: map[string]*logproto.Stream{},
+			},
+			expectedStream: `{__aws_cloudwatch_log_group="testLogGroup", __aws_cloudwatch_owner="123456789123", __aws_log_type="cloudwatch"}`,
+			keepStream:     false,
+		},
+		{
+			name: "cloudwatch_keepStream",
+			b: &batch{
+				streams: map[string]*logproto.Stream{},
+			},
+			expectedStream: `{__aws_cloudwatch_log_group="testLogGroup", __aws_cloudwatch_log_stream="testLogStream", __aws_cloudwatch_owner="123456789123", __aws_log_type="cloudwatch"}`,
+			keepStream:     true,
+		},
+	}
+
+	for _, tt := range tests {
+		// https://github.com/aws/aws-lambda-go/blob/main/events/cloudwatch_logs_test.go
```

Comment:
is this where the `Data` value is copied from? if so can we clarify that

Reply:
Sure, done in 5393735.